builtin/providers/azure/resource_azure_local_network.go

```diff
@@ -0,0 +1,270 @@
+package azure
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/management/virtualnetwork"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceAzureLocalNetworkConnetion returns the schema.Resource associated to an
+// Azure hosted service.
+func resourceAzureLocalNetworkConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAzureLocalNetworkConnectionCreate,
+		Read:   resourceAzureLocalNetworkConnectionRead,
+		Update: resourceAzureLocalNetworkConnectionUpdate,
+		Exists: resourceAzureLocalNetworkConnectionExists,
+		Delete: resourceAzureLocalNetworkConnectionDelete,
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: parameterDescriptions["name"],
+			},
+			"vpn_gateway_address": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: parameterDescriptions["vpn_gateway_address"],
+			},
+			"address_space_prefixes": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: parameterDescriptions["address_space_prefixes"],
+			},
+		},
+	}
+}
+
+// sourceAzureLocalNetworkConnectionCreate issues all the necessary API calls
+// to create a virtual network on Azure.
+func resourceAzureLocalNetworkConnectionCreate(d *schema.ResourceData, meta interface{}) error {
+	azureClient, ok := meta.(*AzureClient)
+	if !ok {
+		return fmt.Errorf("Failed to convert to *AzureClient, got: %T", meta)
+	}
+	managementClient := azureClient.managementClient
+	networkClient := virtualnetwork.NewClient(managementClient)
+
+	log.Println("[INFO] Fetching current network configuration from Azure.")
+	azureClient.mutex.Lock()
+	defer azureClient.mutex.Unlock()
+	netConf, err := networkClient.GetVirtualNetworkConfiguration()
+	if err != nil {
+		return fmt.Errorf("Failed to get the current network configuration from Azure: %s", err)
+	}
+
+	// get provided configuration:
+	name := d.Get("name").(string)
+	vpnGateway := d.Get("vpn_gateway_address").(string)
+	var prefixes []string
+	if nprefixes := d.Get("address_space_prefixes.#").(int); nprefixes > 0 {
+		prefixes = []string{}
```

Comment:
this line is unnecessary; `append` will properly handle the nil slice declared above